torch/csrc/jit/frontend/script_type_parser.cpp

```diff
@@ -217,6 +217,33 @@ TypePtr ScriptTypeParser::parseType(const std::string& str) {
   return parseTypeFromExpr(p.parseExp());
 }
 
+bool checkMutableFunctionDefault(const IValue& def) {
+  if (def.isList() || def.isGenericDict()) {
```

Comment:
can we invoke the existing logic here instead of copy-pasta ? 

https://github.com/pytorch/pytorch/blob/fcc10d75e1a0c1c2474de559124620ca6b580313/torch/csrc/jit/python/script_init.cpp#L223